text/0000-cargo-tags.md

```diff
@@ -0,0 +1,126 @@
+
+Categories and keywords for crates are now deprecated, replaced by a unified "tags" instead. A new set of policies is added to allow the crates.io team to curate the way tags are presented, replacing features such as the "Popular Categories" list on crates.io with a "Popular Tags" instead.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Cargo supports two types of metadata when tagging the purpose/scope of a crate:
+
+1. Categories, which are restricted to a curated set of allowed categories.
+2. Keywords, which are unrestricted and user-defined.
+
+However, the reality is that the actual curation of categories is extremely loose, and the only process for adding a category is to open a PR for the crates.io repository. The required review is effectively asking for an "LGTM" reply from one of the crates.io team members, as seen in [the history for the `categories.toml` file on the repository](https://github.com/rust-lang/crates.io/commits/main/src/boot/categories.toml).
+
+Ultimately, the most important distinction between categories and keywords has really been that the crates.io team has control over the presentation of categories, although a more important distinction is that categories cannot be removed, only added.
+
+Recently, [a discussion](https://github.com/rust-lang/crates.io/discussions/6762) was opened in the crates.io repository on whether the cryptocurrencies category should be removed from crates.io, due to the plethora of issues surrounding them. This should not be treated as a reason for adopting the RFC (although it was a motivation to write it), but instead as something that brought up the fact that categories cannot be removed by policy.
+
+Because categories cannot be removed, they also cannot be renamed or otherwise curated by the community. However, by switching to keywords, we can effectively solve this problem; community members can simply start publishing their crates under different keywords and older, unsupported crates wouldn't have any issue remaining published under their older versions.
+
+To make the change clearer, the unified categories and keywords are called "tags" instead, and any crates using keywords or categories in their manifests will implicitly add those strings to tags instead. Ultimately, switching to a unified, unrestricted system solves a couple problems:
+
+1. Instead of being gatekept by a PR to the crates.io repository, categories can organically be adopted by community members in the form of keywords. Which keywords are most popular and useful can be decided organically.
+2. Keywords can still be given descriptions and other metadata on crates.io, although no distinction between these "special" keywords and other keywords is made in cargo itself. This allows making changes to the way crates are presented without having to worry about backwards compatibility.
+3. Adding, removing, and modifying the curated set of keywords is no longer a technical choice, but a cultural one.
```

Comment:
Since "tags" essentially would work the same way as the current "keywords" concept, I'd prefer to remove "categories" and keep "keywords" instead of removing both and introducing another thing. This would significantly lower the work for the crates.io team, but also for all of the crate authors that would have to adjust their manifests.

Reply:
That's very fair. To be honest, that was how I had it in the initial version of the RFC, but I decided to change it since I thought it might be less confusing to emphasise them as a new type of data altogether.

Depending on how people feel, I'm fine moving back to that version instead.

Reply:
I agree that if we go forward with this that we should just maintain continuity with keywords

Reply:
This has been adopted into the latest version of the RFC.